app/controllers/application_controller.rb

```diff
@@ -60,7 +60,7 @@ def enable_embedded_shopfront
     return if embedding_without_https?
 
     response.headers.delete 'X-Frame-Options'
-    response.headers['Content-Security-Policy'] = "frame-ancestors #{embedded_shopfront_referer}"
+    response.headers['Content-Security-Policy'] = "frame-ancestors #{URI(request.referer).host.downcase}"
```

Comment:
This will work, because the referer got tested for existence before. But it would be nice to have a proper method here.

```ruby
  def csp_frame_ancestors
    return "none" if request.referer.blank?
    URI(request.referer).host.downcase
  end
```